jcl/src/java.base/share/classes/java/lang/StringBuilder.java

```diff
@@ -2843,24 +2826,24 @@ public int offsetByCodePoints(int start, int codePointCount) {
 }
 
 /**
- * Adds the specified code point to the end of this StringBuffer.
+ * Adds the specified code point to the end of this StringBuilder.
  *
  * @param		codePoint	the code point
- * @return		this StringBuffer
+ * @return		this StringBuilder
  */
 public StringBuilder appendCodePoint(int codePoint) {
 	if (codePoint >= 0) {
 		if (codePoint < 0x10000) {
 			return append((char)codePoint);
 		} else if (codePoint < 0x110000) {
+			int currentLength = lengthInternal();
+			int currentCapacity = capacityInternal();
+
 			// Check if the StringBuilder is compressed
 			if (String.COMPACT_STRINGS && count >= 0) {
-				decompress(value.length);
+				decompress(currentLength, currentLength);
 			}
```

Comment:
Please move this after the overflow check, and avoid the call to `ensureCapacityImpl()` when unnecessary.

Reply:
Made the changes requested.